src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizer.cs

```diff
@@ -249,15 +251,54 @@ private string GetResourceStreamName(CultureInfo culture)
             return resourceStreamName;
         }
 
+        private const string AssemblyElementDelimiter = ", ";
+
+        private string ApplyCultureToAssembly(string assemblyFullName, CultureInfo culture)
```

Comment:
Looks like a very allocation-way to insert `.resources` after the assembly name and optionally to add or update the culture. Result will be cached once you've applied my suggestions below. But still, ...

Also needs comments.